recipes/MEDIA/ASR/seq2seq/train_with_wav2vec.py

```diff
@@ -0,0 +1,378 @@
+The system employs a wav2vec model and a decoder.
+
+To run this recipe, do the following:
+> python train_with_wav2vec.py hparams/train_with_wav2vec.yaml
+
+With the default hyperparameters, the system employs a VanillaNN decoder.
+
+The neural network is trained on greedy CTC.
+
+The experiment file is flexible enough to support a large variety of
+different systems. By properly changing the parameter files, you can try
+different encoders, decoders,
+training tasks (Media , PortMedia), and many other possible variations.
+
+Authors
+ * Gaelle Laperriere
+"""
+
+import sys
+import torch
+import logging
+import speechbrain as sb
+from hyperpyyaml import load_hyperpyyaml
+from speechbrain.dataio.batch import PaddedBatch
+from speechbrain.utils.distributed import run_on_main
+from parseXMLtoSB import prepare_media
+
+logger = logging.getLogger(__name__)
+
+
+# Define training procedure.
+class ASR(sb.core.Brain):
+    def compute_forward(self, wavs, wav_lens, stage):
+        """Forward computations from waveform to output probabilities."""
+
+        # Forward pass.
+        feats = self.modules.wav2vec(wavs)
+
+        x = self.modules.dec(feats)
+
+        # Output layer for seq2seq log-probabilities.
+        logits = self.modules.output_lin(x)
+        p_ctc = self.hparams.softmax(logits)
+
+        return p_ctc, wav_lens
+
+    def compute_objectives(self, predictions, ids, chars, char_lens, stage):
+        """Computes the loss (CTC+NLL) given predictions and targets."""
+
+        # Get predictions & loss.
+        p_ctc, wav_lens = predictions
+        loss = self.hparams.ctc_cost(p_ctc, chars, wav_lens, char_lens)
+
+        # Get metrics.
+        if stage != sb.Stage.TRAIN:
+            # Generate sequences with CTC greedy decoder.
+            sequence = sb.decoders.ctc_greedy_decode(
+                p_ctc, wav_lens, self.hparams.blank_index
+            )
+            # Update metrics.
+            self.cer_metric.append(
+                ids=ids,
+                predict=sequence,
+                target=chars,
+                target_len=char_lens,
+                ind2lab=self.label_encoder.decode_ndim,
+            )
+            self.ctc_metric.append(ids, p_ctc, chars, wav_lens, char_lens)
+
+        return loss
+
+    def init_optimizers(self):
+        """Initializes the wav2vec2 optimizer and model optimizer"""
+
+        # Join optimizers.
+        self.optimizer_wav2vec = self.hparams.opt_class_wav2vec(
+            self.hparams.model_wav2vec.parameters()
+        )
+        self.optimizer = self.hparams.opt_class(self.hparams.model.parameters())
+
+        # Add opitmizers to checkpoint recoverables.
+        if self.checkpointer is not None:
+            self.checkpointer.add_recoverable(
+                "optimizer_wav2vec", self.optimizer_wav2vec
+            )
+            self.checkpointer.add_recoverable("optimizer", self.optimizer)
+
+    def fit_batch(self, batch):
```

Comment:
You don't need this function, this is already computed for you here: https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/core.py#L882